src/app/tripManagement/components/trip-details/itinerary-item/itinerary-item.component.ts

```diff
@@ -22,7 +22,7 @@ export class ItineraryItemComponent {
   trip: ITrip | undefined = undefined;
   tripSubscription = new Subscription();
 
-  ngOnInit() {
+  constructor() {
     this.tripSubscription = this.trip$.subscribe((trip) => {
```

Comment:
Just a suggestion. Another way to unsubscribe would be to use `takeUntilDestroyed` from `rxjs-interop` in this way: `this.trip$.pipe(takeUntilDestroyed()).subscribe{...}`. This is, however, just a suggestion, what you have is also  correct.

Reply:
Thank you, yea Hugo also suggested the same too. I've noted it and will get to it if i have time to polish the code at the end